road/scatter.py

```diff
@@ -0,0 +1,236 @@
+import os
+import json
+import yaml
+import pandas as pd
+from sklearn.linear_model import LinearRegression
+
+
+def compute_and_save_errors(
+        est_df,
+        obs_df,
+        chosen_timeperiod,
+        combined_df_cols,
+        classification_col):
+    # Compute errors
+    times = ['Daily', 'AM', 'MD', 'PM', 'EV', 'EA']
+    error = {col: est_df[col] - obs_df[col] for col in times}
+    error_df = pd.DataFrame(error)
+    error_squared_df = error_df.pow(2)
+    error_percent_df = error_df.divide(obs_df[times]).fillna(0)
+    error_percent_df = error_percent_df.apply(lambda x: x * 100, axis=1)
+
+    # Combine dataframes
+    combined_df = pd.concat(
+        [est_df, obs_df[times], error_df, error_squared_df, error_percent_df], axis=1)
+    Select_time_period_df = combined_df.filter(like=chosen_timeperiod)
+    specific_columns_combined_df = combined_df[combined_df_cols]
+    Select_time_period_loc_df = pd.concat(
+        [specific_columns_combined_df, Select_time_period_df], axis=1)
+
+    # Set column names
+    calculation_cols = [
+        'Estimated Volume',
+        'Observed Volume',
+        'Errors',
+        'Squared Errors',
+        'Percent Errors']
+    Select_time_period_loc_df.columns = combined_df_cols + calculation_cols
+
+    # Drop duplicates
+    Select_time_period_loc_df = Select_time_period_loc_df.drop_duplicates(
+        subset=['A', 'B'], keep='first')
+
+    # Save to CSV
+    file_name = f"{chosen_timeperiod}_Scatter_data.csv"
+    Select_time_period_loc_df.to_csv(file_name, index=False)
+
+    # Get classification column types
+    classification_col_types = Select_time_period_loc_df[classification_col].unique(
+    )
+
+    return Select_time_period_loc_df, classification_col_types, file_name
+
+
+def sanitize_filename(filename):

```

Comment:
this shouldn't be needed. Just use `from pathlib import Path` (which can also replace `os.path.join()` too.
https://docs.python.org/3/library/pathlib.html (see the examples in the first section, and a one-to-one matching table at the very end for pathlib vs os.path functions)

In general, don't hardcode Windows vs *NIX filepath conventions into the code (though I know that we only use Windows here)